elastic-utils-lib/src/cmr/elastic_utils/index_util.clj

```diff
@@ -213,3 +214,16 @@
   "Creates the alias for the index."
   [conn index alias]
   (esi/update-aliases conn [{:add {:index index :alias alias}}]))
+
+(spec/def ::elasticsearch-index-stats
+  (spec/keys :req-un [::indices]
+             :opt-un [::_shards ::_all ::error ::status]))
+
+(defn index-exists?
+  "Returns whether an index with a given name exists in Elasticsearch."
+  [conn index-name]
+  (let [stats (esi/stats conn index-name)]
+    (and (spec/valid? ::elasticsearch-index-stats stats)
+         ;; status 404 is only given when not found, status is not normally returned
+         (not= 404 (:status stats)))))
+
```

Comment:
Just curious what should happen when the index exists but it contains error?  In which case, do you or do you not want to do autocomplete indexing?

Reply:
If the index exists but there is an error, I would rather Elasticsearch itself fail during an ingest and have that error be available in the logs.